### DISCUSSION TITLE
Differences between community.general.keycloak* and this collection

### DISCUSSION THREAD

**Discussion by martux69** [ansible-middleware/keycloak#246] [General]
What are the differences between these collection and which is the preferred one?
https://docs.ansible.com/ansible/latest/collections/community/general/keycloak_authentication_module.html#ansible-collections-community-general-keycloak-authentication-module

**Comment by guidograzioli** [comment]
- community.general has a python client for the keycloak API, and modules using it, each of calling into API endpoints.

- middleware_automation.keycloak provides roles to install and configure the keycloak service; additionally, it provides a role to configure a keycloak realm, which makes use of modules forked from the  community_general keycloak_ modules. 

So, for configuring via task/modules you'd choose community.general (if you can use it); otherwise, you can use keycloak_realm role from this collection.

**Comment by guidograzioli** [comment]
More discussion on this topic happening here:
https://forum.ansible.com/t/keycloak-modules-in-community-general/41746
